gameplan/extends/client.py

```diff
@@ -5,21 +5,36 @@
 import frappe
 from frappe.model.base_document import get_controller
 
-
 @frappe.whitelist()
-def get_list(doctype=None, fields=None, filters=None, order_by=None, start=0, limit=20, group_by=None, parent=None, debug=False):
-	check_permissions(doctype, parent)
-	query = frappe.qb.get_query(
-		table=doctype,
-		fields=fields,
-		filters=filters,
-		order_by=order_by,
-		offset=start,
-		limit=limit,
-		group_by=group_by,
+def get_list(doctype, **kwargs):
+	controller = get_controller(doctype)
+	if hasattr(controller, "get_list_query"):
+		check_permissions(doctype, kwargs.get("parent"))
```

Comment:
I hope scoping this logic to what is supposed to be a gp-specific controller, only is the right thing to do an gp didn't depend on generally modified check_permissions. :pray: 

If that's the case, it makes things a bit more difficult of course. In that case, all gp requests could send a sentinel value to help trigger (or not) that logic.